[Bug]: A5不支持cudagraph_mode设置为“FULL”

### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
Your output of above commands here
```

</details>
场景：A5单机混部场景

服务部署脚本：
```
export ASCEND_RT_VISIBLE_DEVICES=4
export PYTORCH_NPU_ALLOC_CONF=expandable_segments:True
export TASK_QUEUE_ENABLE=1
export VLLM_ASCEND_ENABLE_PREFETCH_MLP=1

vllm serve /home/weights/qwen3-32B-w8a8-no-w_axes-1118-full \
  --served-model-name qwen3 \
  --trust-remote-code \
  --distributed-executor-backend mp \
  --tensor-parallel-size 1 \
  --max-model-len 5500 \
  --max-num-batched-tokens 40960 \
  --no-enable-prefix-caching \
  --async-scheduling \
  --quantization ascend \
  --compilation-config '{"cudagraph_mode": "FULL"}' \
  --additional-config '{"pa_shape_list":[32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255,256]}'\
  --speculative_config '{"method": "eagle3", "model":"/home/weights/Qwen3-32B-speculator.eagle3", "enforce_eager": true, "num_speculative_tokens": 3}' \
  --host 141.61.41.214 \
  --port 2000 \
  --block-size 128 \
  --gpu-memory-utilization 0.9 2>&1 | tee FULL.log
```
报错信息：
```
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/vllm/compilation/piecewise_backend.py", line 367, in __call__
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     return range_entry.runnable(*args)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/torch/_functorch/aot_autograd.py", line 1130, in forward
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     return compiled_fn(full_args)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]            ^^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/torch/_functorch/_aot_autograd/runtime_wrappers.py", line 353, in runtime_wrapper
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     all_outs = call_func_at_runtime_with_args(
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/torch/_functorch/_aot_autograd/utils.py", line 129, in call_func_at_runtime_with_args
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     out = normalize_as_list(f(args))
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]                             ^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/torch/_functorch/_aot_autograd/runtime_wrappers.py", line 724, in inner_fn
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     outs = compiled_fn(args)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]            ^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/torch/_functorch/_aot_autograd/runtime_wrappers.py", line 526, in wrapper
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     return compiled_fn(runtime_args)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]            ^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/torch/_functorch/_aot_autograd/utils.py", line 103, in g
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     return f(*args)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]            ^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/torch_npu/dynamo/torchair/npu_fx_compiler.py", line 525, in __call__
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     gm_result = self.run_kernel(*args, **kwargs)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "<string>", line 2963, in kernel
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/torch_npu/dynamo/torchair/_acl_concrete_graph/acl_graph.py", line 830, in __call__
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     return self.fx_run_eagerly(*args, **kwargs)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/torch_npu/dynamo/torchair/_acl_concrete_graph/acl_graph.py", line 901, in fx_run_eagerly
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     return self.fx_forward(*args, **kwargs)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "<string>", line 40, in forward
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/torch/_ops.py", line 841, in __call__
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     return self._op(*args, **kwargs)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]            ^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/vllm/model_executor/layers/attention/kv_transfer_utils.py", line 39, in wrapper
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     return func(*args, **kwargs)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]            ^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/vllm/model_executor/layers/attention/attention.py", line 720, in unified_attention_with_output
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     self.impl.forward(
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/y00899271/pkgs_0413/vllm-ascend/vllm_ascend/attention/attention_v1.py", line 1059, in forward
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     attn_output = self.forward_impl(query, key, value, kv_cache, attn_metadata, output_padded)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/y00899271/pkgs_0413/vllm-ascend/vllm_ascend/attention/attention_v1.py", line 997, in forward_impl
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     output = self.forward_fused_infer_attention(query, key, value, attn_metadata, output, kv_cache)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/y00899271/pkgs_0413/vllm-ascend/vllm_ascend/attention/attention_v1.py", line 868, in forward_fused_infer_attention
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     attn_output, _ = torch_npu.npu_fused_infer_attention_score(
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/torch/_ops.py", line 1255, in __call__
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     return self._op(*args, **kwargs)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]            ^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] RuntimeError: npu_fused_infer_attention_score_symint:../third_party/op-plugin/op_plugin/ops/opapi/FusedInferAttentionScoreKernelNpuOpApi.cpp:437 NPU function error: device error type 0, error code is 561002
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] [ERROR] 2026-05-14-09:24:36 (PID:2497238, Device:0, RankID:-1) ERR00100 PTA call acl api failed.
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] EZ9999: Inner Error!
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] EZ9999[PID: 2497238] 2026-05-14-09:24:36.504.007 (EZ9999):  The size(256) of actualSeqLengthsKv is not equal to the batchSize(257). The size of actualSeqLengthsKv must be equal to the batchSize when the layout of key/value is TND or NTD.[FUNC:CheckActualSeqLenKvDim][FILE:actual_seq_len_checker.cpp][LINE:156]
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]         TraceBack (most recent call last):
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Execute FusedInferAttentionScore tiling function failed, ret is -1. Please check your implementation of tiling function.
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Check NnopbaseExecutorDoTiling(executor) failed
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Check NnopbaseExecutorTilingAndUpdateBinInfo(executor) failed
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Check NnopbaseExecutorMatchCache(executor) failed
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Check NnopbaseRunForWorkspace(*executor, workspaceSize) failed
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] 
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] 
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] During handling of the above exception, another exception occurred:
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] 
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] Traceback (most recent call last):
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/y00899271/pkgs_0413/vllm-ascend/vllm_ascend/worker/model_runner_v1.py", line 3610, in _torch_cuda_wrapper
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     yield
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/y00899271/pkgs_0413/vllm-ascend/vllm_ascend/worker/model_runner_v1.py", line 3537, in capture_model
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     with _torch_cuda_wrapper(), _replace_gpu_model_runner_function_wrapper(parent_module_name):
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/contextlib.py", line 158, in __exit__
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     self.gen.throw(typ, value, traceback)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/y00899271/pkgs_0413/vllm-ascend/vllm_ascend/worker/model_runner_v1.py", line 3639, in _replace_gpu_model_runner_function_wrapper
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     raise RuntimeError(f"NPUModelRunner failed, error is {e}")
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] RuntimeError: NPUModelRunner failed, error is npu_fused_infer_attention_score_symint:../third_party/op-plugin/op_plugin/ops/opapi/FusedInferAttentionScoreKernelNpuOpApi.cpp:437 NPU function error: device error type 0, error code is 561002
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] [ERROR] 2026-05-14-09:24:36 (PID:2497238, Device:0, RankID:-1) ERR00100 PTA call acl api failed.
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] EZ9999: Inner Error!
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] EZ9999[PID: 2497238] 2026-05-14-09:24:36.504.007 (EZ9999):  The size(256) of actualSeqLengthsKv is not equal to the batchSize(257). The size of actualSeqLengthsKv must be equal to the batchSize when the layout of key/value is TND or NTD.[FUNC:CheckActualSeqLenKvDim][FILE:actual_seq_len_checker.cpp][LINE:156]
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]         TraceBack (most recent call last):
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Execute FusedInferAttentionScore tiling function failed, ret is -1. Please check your implementation of tiling function.
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Check NnopbaseExecutorDoTiling(executor) failed
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Check NnopbaseExecutorTilingAndUpdateBinInfo(executor) failed
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Check NnopbaseExecutorMatchCache(executor) failed
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Check NnopbaseRunForWorkspace(*executor, workspaceSize) failed
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] 
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] 
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] During handling of the above exception, another exception occurred:
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] 
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] Traceback (most recent call last):
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/site-packages/vllm/v1/executor/multiproc_executor.py", line 944, in worker_busy_loop
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     output = func(*args, **kwargs)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]              ^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/y00899271/pkgs_0413/vllm-ascend/vllm_ascend/worker/worker.py", line 498, in compile_or_warm_up_model
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     npugraph_memory_bytes = self.model_runner.capture_model()
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/y00899271/pkgs_0413/vllm-ascend/vllm_ascend/worker/model_runner_v1.py", line 3537, in capture_model
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     with _torch_cuda_wrapper(), _replace_gpu_model_runner_function_wrapper(parent_module_name):
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/miniconda3/envs/yxr_vllm/lib/python3.11/contextlib.py", line 158, in __exit__
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     self.gen.throw(typ, value, traceback)
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]   File "/home/y00899271/pkgs_0413/vllm-ascend/vllm_ascend/worker/model_runner_v1.py", line 3619, in _torch_cuda_wrapper
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]     raise RuntimeError(f"NPUModelRunner init failed, error is {e}")
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] RuntimeError: NPUModelRunner init failed, error is NPUModelRunner failed, error is npu_fused_infer_attention_score_symint:../third_party/op-plugin/op_plugin/ops/opapi/FusedInferAttentionScoreKernelNpuOpApi.cpp:437 NPU function error: device error type 0, error code is 561002
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] [ERROR] 2026-05-14-09:24:36 (PID:2497238, Device:0, RankID:-1) ERR00100 PTA call acl api failed.
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] EZ9999: Inner Error!
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] EZ9999[PID: 2497238] 2026-05-14-09:24:36.504.007 (EZ9999):  The size(256) of actualSeqLengthsKv is not equal to the batchSize(257). The size of actualSeqLengthsKv must be equal to the batchSize when the layout of key/value is TND or NTD.[FUNC:CheckActualSeqLenKvDim][FILE:actual_seq_len_checker.cpp][LINE:156]
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]         TraceBack (most recent call last):
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Execute FusedInferAttentionScore tiling function failed, ret is -1. Please check your implementation of tiling function.
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Check NnopbaseExecutorDoTiling(executor) failed
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Check NnopbaseExecutorTilingAndUpdateBinInfo(executor) failed
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Check NnopbaseExecutorMatchCache(executor) failed
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949]        Check NnopbaseRunForWorkspace(*executor, workspaceSize) failed
(Worker pid=2497238) ERROR 05-14 09:24:37 [multiproc_executor.py:949] 
```

### 🐛 Describe the bug

A5不支持cudagraph_mode设置为“FULL”

--- a/vllm_ascend/worker/model_runner_v1.py
+++ b/vllm_ascend/worker/model_runner_v1.py
@@ -645,9 +645,11 @@ class NPUModelRunner(GPUModelRunner):
             # Mixed-batch case: num_reqs must equal num_reqs_padded
             assert num_reqs == num_reqs_padded
 
-            # Insert a dummy request instead of setting query_start_loc[num_reqs] = num_tokens_padded directly
-            self.query_start_loc.np[num_reqs_padded + 1] = num_tokens_padded
-            num_reqs_padded = num_reqs_padded + 1
+            # Do not insert if the last value already equals the num_tokens
+            if self.query_start_loc.np[num_reqs_padded] < num_tokens_padded:
+                # Insert a dummy request instead of change the last value directly
+                self.query_start_loc.np[num_reqs_padded + 1] = num_tokens_padded
+                num_reqs_padded = num_reqs_padded + 1
 
         self.query_start_loc.copy_to_gpu()
 

